### DISCUSSION TITLE
Lunar major ArchaeoLines: what do they mean

### DISCUSSION THREAD

**Discussion by vreijs** [Stellarium/stellarium#3331] [Q&A]
Dear ArchaeoLines developer,

What is the difference between the two 'Major Lunar Standstill' in ArchaeoLines?
Reading the About it has to do with the nearest and furthest Moon distance. Correct?

I wonder if it could be improved somewhat? I would think that a topocentric declination line for the Moon at transit and a Moon at apparent altitude =0deg would be handy (these two thus provide the variation of parallax due to distance and apparent altitude).

Some examples
For northern events
Moon at transit (on northern major standstill event date for transit):
![image](https://github.com/Stellarium/stellarium/assets/17218070/13fe8df2-9c48-4791-80e2-e0bb765eca59)
Moon at apparent alt=0deg  (on northern major standstill event date for set):
![image](https://github.com/Stellarium/stellarium/assets/17218070/7a7973c6-0593-4554-8b4c-6b4d69a94600)
<The Moon needs to be below these two limiting lines and the lines show the 'theoretical' topocentric limit>.

For southern events
Moon at transit (on southern major standstill event date for transit):
![image](https://github.com/Stellarium/stellarium/assets/17218070/0a9c0eb1-44c6-4623-824e-e0c1568806f5)
Moon at apparent alt=0deg  (on southern major standstill event date for set):
![image](https://github.com/Stellarium/stellarium/assets/17218070/7c9a6c74-4750-4f9c-80c9-25f2883c27ce)
<The Moon needs to be above these two limiting lines and the lines show the 'theoretical' topocentric limit>.

As you can see the Moon is not really near the 'Major Lunar Standstill' lines (from ArchaeoLines), but closer to the '2024 transit or horizon' lines (Custom lines).
<the topocentric declinations are calculated using ARCHAEOCOSMO's lunar limit function, which depends on JDN date,  type of standstill event (northern/southern), Altitude, Latitude>

It would even be nicer if the lunar (topocentric) lines in Stellarium would depend on the latitude en altitude; aka the parallax. But this might be a little more complicated than the two lines depicting transit and horizon,

Anyway, it is an idea which can help I hope.
All the best,

Victor

**Comment by gzotti** [comment]
Hi Victor!

In the Guide I have written:
>For the lunar events, there are two lines each drawn by the plugin, for maximum and minimum
distance of the moon. The lunar extreme declinations are computed taking horizon parallax effects
into account. For technical reasons however, the derived declinations are then used to draw small
circles of constant declinations on the sphere, without taking the change of lunar horizontal parallax
into account. Note that therefore the observed declination of the moon at the major standstill can
exceed the indicated limits if it is high in the sky. The main purpose of this plugin is however to
show an indication of the intersection of the standstill lines with the horizon.

From these descriptions, I hope it is clear: I find horizontal parallax from location, extreme declination (&pm;&epsilon;&pm;_i_ with _i_=5.145396° regarded constant. You may argue that _i_ is variable by a few arcminutes with ~1/2 year period, I don't include that either), and theoretical min/max distance of the Moon, compute apparent declination on the intersection with the mathematical horizon, and then draw a line of constant declination (only affected by refraction). Above the horizon the parallax effect changes and therefore does not show the theoretical limit at transit, although it is close enough for most purposes. Given that the physical Moon never follows a line of constant declination over the sky, other refinements are not planned, and I am afraid adding yet more lines e.g. for any theoretical apparent min/max declination at transit rather adds to confusion.  I could add a few more custom declination lines. How many do you need? 3? 4? 5? 10? 20? You are one of probably 15 "targeted users" for this plugin, and maybe the first one to ever need more than 2 lines. I can envision that it may even be useful to spawn a "current object declination" line for the Moon at a particular moment and then adjust it to cover its northern and southern limits, as required by the question under investigation. There will always be more ideas than time to do everything. 

What would be nice is if the Lunar theories or positional algorithms would allow deriving nodal longitudes to give immediate indication of the standstill instance. They are not prepared for this. Observe ecliptic, ecl. longitude of Earth shadow and lunar orbit line.

**Comment by vreijs** [comment]
Hello Georg,

I see two as enough. But with this definition that I gave. The maximum and
minimum distance of the Moon are not that helpful (at least not for me).
The two I propose give that plus the variation due to apparent altitude.
No problem if this is done somewhere in the future. I can understand time
is a problem.
I can myself determine the topocentric declinations due to parallax (caused
by altitude *and* distance) and input them in the Custom lines (I will
uncheck the 'Major Lunar Standstill' lines).

All the best,

Victor

**Comment by gzotti** [comment]
Horizontal parallax is of course included. What else is computed is in the source code for everybody to see. I accept pull requests.

Your last image seems to indicate that I should somehow add that mentioned 8 arcminute fluctuation of _i_. The Moon is close to apogee, so the lower of the two lines applies. Pushing that further down a few arcminutes (as appropriate to a half-moon during major standstill) leads me to the green "current" line. As said, I have no direct information of the current Lunar orbit quantities, so we must live with the average limits.

**Reply by vreijs** [reply]
I might have a look at the code: https://github.com/Stellarium/stellarium/blob/master/plugins/ArchaeoLines/src/ArchaeoLines.cpp

I use my own determination of the limiting declinations and it seems to be quite good (compared to the actual values; there will always be some difference). 
About adding the perturbation (8' 7": http://www.archaeocosmology.org/eng/moonperb.htm ), did you not include that already? That is the most logical one to include for determining the limiting lunar (geocentric) declinations (beside the inclination). The perturbation is not really related to parallax (parallax is a separate issue to get to topocentric coordinates).
I am now auditing the literature topocentric declinations of Calanais I's skyline reappearing events with Stellarium 3D scenery and for that I need relative ok-ish lunar limits (although I seem to have problems with the low resolution skylines made with STRM 1"). I can of course include my own ones in the 'Custom Lines', but it would be easier if the one that are in ArchaeoLines are closer to the real theoretical limits. 
I will not do any pulls, sorry, but I might propose code changes. Will have a look if I see something obvious.

All the best,

Victor

**Reply by gzotti** [reply]
As I have written, we have no notion about orbital perturbations which could be derived from the modern style of JD->(x,y,z) position deriving method (ELP-82 or DE430/431/440/441), and therefore I have no 8' perturbation in the estimate of a theoretical limit line. However, given its geometrical (3D) nature, we DO have parallax correction. I could add a constant 8' to the values of _i_ for the 'Major' lines. And what to do for the 'Minor'? Or shall I split each line into 2 more? I don't see the real decisive value in it, but I could add a line on the topic to the Guide.

**Reply by vreijs** [reply]
I would not add any more lines, just the correct limiting line(s) (aka closest to the limits, one would be great, but then including apparent altitude for parallax is more difficult to include I assume [I have it included in my limit calculations, I can provide my VBA code, I might even have it in R...)... Knowledge of this perturbation, etc is just internal (and not important for the normal users). 
The perturbation is a very simple geometric value (as you can see on my given web page; which provide ideas for major and minor limits). It is even simpler than parallax, IMHO (which is depending on latitude and apparent altitude).
Again, I will check your code and see what it easy and I think perturbation is very easy (I am not interested in difficult things).

**Comment by vreijs** [comment]
This is what I do to include the perturbation:
MoonInclination = 5.1454
MoonPerturb = 0.145
GeoDeclfromSolarLunarEvent = Sunobliquity(JDNDays)
If Object = "moonmajor" Then GeoDeclfromSolarLunarEvent =
GeoDeclfromSolarLunarEvent + (MoonInclination + MoonPerturb)
If Object = "moonminor" Then GeoDeclfromSolarLunarEvent =
GeoDeclfromSolarLunarEvent - (MoonInclination + MoonPerturb)

<this is before adding the parallax>
see also Meeus, _Mathematical astronomy morsels_, 1997, page 27

All the best,

Victor

**Reply by gzotti** [reply]
Alright. Now that we have completed "Algorithms", it's time to go through the "Morsels". :-)

**Comment by vreijs** [comment]
By the way, if you only want to display the limiting major/minor lines,
then you could choose for nearest or furthest in the following way:

' NS (0=South-Winter,1=North-Summer)
  If NS = 0 Then
        GeoDeclfromSolarLunarEvent = GeoDeclfromSolarLunarEvent -
ParDecfromGeoAltLat(Lat, GeoAlt, 0, GeoDeclfromSolarLunarEvent,
"moonnearest")
    Else
        GeoDeclfromSolarLunarEvent = GeoDeclfromSolarLunarEvent -
ParDecfromGeoAltLat(Lat, GeoAlt, 0, GeoDeclfromSolarLunarEvent,
"moonfurthest")
    End If

<forget the dependency of my parallax function on on GeoAlt, Lat etc; the
lunar distance used is depending on the southern or northern event>

But keeping the two lines is of course also possible (although I am more
interested in the limiting value, but others might want to know more about
the 'internals'>

**Reply by vreijs** [reply]
Also checking the parallax angles you are using (I understand at topocentric altitude of zero degrees: aka topocentric rise/set).
You are using Meeus 1998 data. If one looks at _More mathematical astronomical Morsels_ 2002 or _Mathematical astronomical morsels III_, 2004. I read: max. apogee of 406720km and minimum perigee of 356353km. This is different than your derived 1998 data (max. apogee of 414632km and minimum perigee of 355369km).
Meeus used ELP-2000/82. Perhaps I should check what DE431 provides as distances.

**Comment by vreijs** [comment]
Also checked the Sun's obliquity. Your getRotObliquity might be a little off:
I use (with Chapront=True):
Public Function Sunobliquity(JDNDays)
' JDNDays [Day]
' Sunobliquity [deg]

    ' used by V. Reijs
    ' http://www.iol.ie/~geniet/eng/moonfluct.htm
    Tbret = DateConversion(JDNDays, "julian", 2000) / 1000
If Chapront Then
' J Hilton et al, Celest.Mech.Dyn.Astron. 94 (2006), 351
    Tbret = 10 * Tbret
    Sunobliquity = (84381.406 + (-46.836769 + (-0.0001831 + (0.0020034 + (-0.000000576 + (-0.0000000434) * Tbret) * Tbret) * Tbret) * Tbret) * Tbret) / 3600
Else
    ' P. Bretagnon, Planetary Programs and tables from -4000 to +2800, 1986, page 6
    Sunobliquity = 23.4392911 - 0.130025833 * Tbret - 0.00000430556 * Tbret ^ 2 + 0.000555347 * Tbret ^ 3 - 0.00000142722 * Tbret ^ 4 - 0.000000693528 * Tbret ^ 5 - 0.0000000108472 * Tbret ^ 6 + 0.000000000197778 * Tbret ^ 7
End If
End Function

My Sunobliquity() result is similar to your Sun's declination seen in Stellarium at the specific times. Your ArchaeoLines are though a little off; only 17", so not that much. But I am sure that my added precision will not causing any problems in Stellarium memory;-)

Hope this helps. Anyway I leave it to you if it is important enough to change.

**Reply by gzotti** [reply]
Hopefully just a different model, implemented in precession.c

>  * Precession solution following methods from:
>  * J. Vondrak, N. Capitaine, P. Wallace
>  * New precession expressions, valid for long time intervals
>  * Astronomy&Astrophysics 534, A22 (2011)
>  * DOI: 10.1051/0004-6361/201117274
>  *   with correction from
>  * J. Vondrak, N. Capitaine, P. Wallace
>  * New precession expressions, valid for long time intervals (Corrigendum)
>  * A&A 541, C1 (2012)
>  * DOI: 10.1051/0004-6361/201117274e
>  * The data are only applicable for a time range of 200.000 years around J2000.

If there is an error in the implementation of this model, I gladly accept hints.

**Reply by vreijs** [reply]
I find it strange that such an accurate formula (of Vondrak) is less accurate around say 2300BCE than an older 5th polynomal . I will do  fast check on the parameters (a typo is sometimes possible with all these numbers).

**Reply by gzotti** [reply]
Who says it's less accurate? What is the reference? But yes, typos are possible.

**Comment by vreijs** [comment]
line 359 		double epsRad=ssystem->getEarth()->getRotObliquity(newJDE);
of https://github.com/Stellarium/stellarium/blob/master/plugins/ArchaeoLines/src/ArchaeoLines.cpp

I don't see getRotObliquity-function in:
https://github.com/Stellarium/stellarium/blob/master/src/core/planetsephems/precession.c

All the best,

Victor

**Reply by vreijs** [reply]
Just to be sure, I don't know if the obliquity formula  I use, is valid for a rang more than 8000years (around 2000CE). At least the DE431 values on the solstice days are very close (within 1") to the obliquity formula value for 2000, -1549 and -2368. Sorry I am not able to verify the validity period of this formula (J Hilton et al, Celest.Mech.Dyn.Astron. 94 (2006), 351).

**Reply by gzotti** [reply]
It's the other way, functions from precession.c are called from Planet.cpp.

**Comment by vreijs** [comment]
I am saying that it (the implementation) looks less accurate and that I
find that strange! DE431 (that was my reference) is close to my implemented
5th polynomial while the implemented Vondrak is slightly different (i know
only a few seconds, but still). So I want to investigate (although it will
have totally no influence on my Calanais I investigations;-)

**Reply by vreijs** [reply]
I see no typos in precession.c. 
Any way I did a better comparison between Stellarium en my own software (partly Swiss Ephemeris and ARCHAEOCOSMO) at winter solstices:
![image](https://github.com/Stellarium/stellarium/assets/17218070/42be5f23-aed9-4ecc-8dd6-72afca7eead3)
So the two DE431 implementations give slightly different solstice declinations of a few arcseconds (the Stellarium values are perhaps plus or minus 1 arcseconds as I used the pointer coordination tool). So close enough (certainly for my archaeocosmology research...
The obliquity formulas provide a slightly different angle (in both cases: Stellarium en ARCHAEOCOSMO) a difference of resp some 15 or 20 arcsecond compared to their respective DE431's winter solstice declinations.  This close enough for my archaeocosmology research...

But in general there is a difference (of some 17arcsec) for the winter solstices declination and the obliquity angle looking at DE431 and obliquity formula. Perhaps another angle (quite constant of over the range 2000CE and 2368BCE) is needed between obliquity and solstice declination.

Anyway it are small differences (and not important form my Calanais I research).

All the best,

Victor

**Reply by gzotti** [reply]
Are you computing nutation as well? I could not find a validity estimate for the IAU2000B model. There may be some arcseconds hidden.

**Comment by vreijs** [comment]
It could very well be nutation (I included parallax to determine solstice
declination from obliquity). The nutation in obliquity is in the order of
around 8 arcsec (but I have no accurate formula...).
